Dockerfile

```
@@ -1,18 +1,22 @@
 FROM golang:1.11-stretch AS builder
-WORKDIR /go/src/github.com/pusher/oauth2_proxy
+
+# Download tools
+RUN wget -O $GOPATH/bin/dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64
+RUN chmod +x $GOPATH/bin/dep
+
+# Copy sources
+WORKDIR $GOPATH/src/github.com/pusher/oauth2_proxy
 COPY . .
 
 # Fetch dependencies
-RUN wget -O dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64
-RUN chmod +x dep
-RUN mv dep $GOPATH/bin/dep
 RUN dep ensure --vendor-only
 
-# Build image
-RUN ./configure && make clean oauth2_proxy
+# Build binary
+RUN ./configure && make build.static && strip oauth2_proxy.static
 
-# Copy binary to debian
-FROM debian:stretch
-COPY --from=builder /go/src/github.com/pusher/oauth2_proxy/oauth2_proxy /bin/oauth2_proxy
+# Copy binary to alpine
+FROM alpine:latest
+COPY --from=builder /go/src/github.com/pusher/oauth2_proxy/oauth2_proxy.static /bin/oauth2_proxy
+COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
```

I'd rather we didn't rely on the copy of this from the builder image and use `apk` to fetch the latest version, can we sub in the below?
```suggestion
RUN apk add --no-cache ca-certificates
```